homeassistant/components/anova/sensor.py

```diff
@@ -105,11 +112,27 @@ async def async_setup_entry(
 ) -> None:
     """Set up Anova device."""
     anova_data: AnovaData = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities(
-        AnovaSensor(coordinator, description)
-        for coordinator in anova_data.coordinators
-        for description in SENSOR_DESCRIPTIONS
-    )
+    valid_entities: set[AnovaSensor] = set()
+    ent_registry = er.async_get(hass)
+    for coordinator in anova_data.coordinators:
+        for description in SENSOR_DESCRIPTIONS:
+            sensor = AnovaSensor(coordinator, description)
+            if f"{DOMAIN}.{SENSOR_DOMAIN}_{sensor.unique_id}" in ent_registry.entities:
+                # If the entity has been added before - we know it is supported.
+                valid_entities.add(sensor)
```

Comment:
Why do we do this?

Why don't remove the non working ones and just setup like normal?

Reply:
So it's less about them not working. This is just a means of adding the existing entities even when the device is offline. Since the entities will exist, when the websocket ends up sending update data - the sensors should automatically update.

Basically - if we have added an entity before - it is supported and we should add it again, even if the device is offline.